packages/gatsby/src/services/types.ts

```diff
@@ -17,6 +17,11 @@ export interface IMutationAction {
 }
 export interface IBuildContext {
   program?: IProgram
+  nodesMutatedDuringQueryRun: boolean
```

Comment:
This lets us check if e.g. a custom resolver has added nodes. If so, we will re-run queries. This is implemented in another machine.